API calls are getting treated as API routes with the `serve` command

## Type of Change
<!-- New Feature / Enhancement / Bug / Other -->

## Summary
In testing **v0.28.0-alpha.1** release saw this error popping up in projects that make `/api/*` HTTP calls.
```sh
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/owenbuckley/Workspace/analogstudios.net/repo/www.analogstudios.net/src/api/v2/events.js' imported from /Users/owenbuckley/Workspace/analogstudios.net/repo/www.analogstudios.net/node_modules/@greenwood/cli/src/plugins/resource/plugin-api-routes.js
    at new NodeError (node:internal/errors:393:5)
    at finalizeResolution (node:internal/modules/esm/resolve:323:11)
    at moduleResolve (node:internal/modules/esm/resolve:916:10)
    at defaultResolve (node:internal/modules/esm/resolve:1124:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:841:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ESMLoader.import (node:internal/modules/esm/loader:525:22)
    at importModuleDynamically (node:internal/modules/esm/translators:110:35)
    at importModuleDynamicallyCallback (node:internal/process/esm_loader:35:14) {
  code: 'ERR_MODULE_NOT_FOUND'
```

Interestingly enough, everything still works though (the events calendar) even though the response in the browser is a `500`🤷‍♂️ 

<img width="1677" alt="Screen Shot 2023-02-08 at 1 20 35 PM" src="https://user-images.githubusercontent.com/895923/217618093-20d67146-245e-4437-8aa7-4adda37d744f.png">



## Details
Not sure what's happening, the API plugin _should_ be guarding against that.

Projects impacted
- https://github.com/AnalogStudiosRI/www.analogstudios.net/pull/85
- https://github.com/ContributaryCommunity/www.contributary.community/pull/121

--- a/packages/cli/src/commands/serve.js
+++ b/packages/cli/src/commands/serve.js
@@ -1,4 +1,5 @@
 import { getStaticServer, getHybridServer } from '../lifecycles/serve.js';
+import { checkResourceExists } from '@greenwood/cli/src/lib/resource-utils.js';
 
 const runProdServer = async (compilation) => {
 
@@ -6,7 +7,7 @@ const runProdServer = async (compilation) => {
 
     try {
       const port = compilation.config.port;
-      const hasApisDir = compilation.context.apisDir;
+      const hasApisDir = await checkResourceExists(compilation.context.apisDir);
       const hasDynamicRoutes = compilation.graph.filter(page => page.isSSR && ((page.data.hasOwnProperty('static') && !page.data.static) || !compilation.config.prerender));
       const server = hasDynamicRoutes.length > 0 || hasApisDir ? getHybridServer : getStaticServer;
 

--- a/packages/cli/src/lifecycles/serve.js
+++ b/packages/cli/src/lifecycles/serve.js
@@ -254,12 +254,16 @@ async function getHybridServer(compilation) {
   app.use(async (ctx) => {
     try {
       const url = new URL(`http://localhost:8080${ctx.url}`);
-      const isApiRoute = url.pathname.startsWith('/api');
       const matchingRoute = compilation.graph.find((node) => node.route === url.pathname) || { data: {} };
       const request = new Request(url.href, {
         method: ctx.request.method,
         headers: ctx.request.header
       });
+      const apiResource = resourcePlugins.find((plugin) => {
+        return plugin.isGreenwoodDefaultPlugin
+          && plugin.name === 'plugin-api-routes';
+      }).provider(compilation);
+      const isApiRoute = await apiResource.shouldServe(url, request);
 
       if (matchingRoute.isSSR && !matchingRoute.data.static) {
         const standardHtmlResource = resourcePlugins.find((plugin) => {
@@ -274,10 +278,6 @@ async function getHybridServer(compilation) {
         ctx.set('Content-Type', 'text/html');
         ctx.status = 200;
       } else if (isApiRoute) {
-        const apiResource = resourcePlugins.find((plugin) => {
-          return plugin.isGreenwoodDefaultPlugin
-            && plugin.name === 'plugin-api-routes';
-        }).provider(compilation);
         const response = await apiResource.serve(url, request);
 
         ctx.status = 200;

--- a/packages/cli/test/cases/serve.default/serve.default.spec.js
+++ b/packages/cli/test/cases/serve.default/serve.default.spec.js
@@ -42,7 +42,7 @@ describe('Serve Greenwood With: ', function() {
     this.context = {
       hostname
     };
-    runner = new Runner();
+    runner = new Runner(true);
   });
 
   describe(LABEL, function() {
@@ -92,6 +92,41 @@ describe('Serve Greenwood With: ', function() {
 
       it('should return the correct response body', function(done) {
         expect(response.body).to.have.lengthOf(1);
+        done();
+      });
+    });
+
+    // https://github.com/ProjectEvergreen/greenwood/issues/1059
+    describe('Serve command with dev proxy with an /api prefix', function() {
+      let response = {};
+
+      before(async function() {
+        return new Promise((resolve, reject) => {
+          request.get(`${hostname}/api/posts?id=7`, (err, res, body) => {
+            if (err) {
+              reject();
+            }
+
+            response = res;
+            response.body = JSON.parse(body);
+
+            resolve();
+          });
+        });
+      });
+
+      it('should return a 200 status', function(done) {
+        expect(response.statusCode).to.equal(200);
+        done();
+      });
+
+      it('should return the correct content type', function(done) {
+        expect(response.headers['content-type']).to.contain('application/json');
+        done();
+      });
+
+      it('should return the correct response body', function(done) {
+        expect(JSON.stringify(response.body)).to.equal('{}');
         done();
       });
     });

--- a/packages/cli/test/cases/serve.default/serve.default.spec.js
+++ b/packages/cli/test/cases/serve.default/serve.default.spec.js
@@ -42,7 +42,7 @@ describe('Serve Greenwood With: ', function() {
     this.context = {
       hostname
     };
-    runner = new Runner(true);
+    runner = new Runner();
   });
 
   describe(LABEL, function() {

